lib/Conversion/TritonGPUToLLVM/ScanOpToLLVM.cpp

```diff
@@ -281,8 +280,16 @@ ScanOpConversion::emitFastScan(triton::ScanOp op, triton::ScanOpAdaptor adaptor,
   if (!helper.isSupported())
     return failure();
 
+  // Obtain global laneId and pass it around
+  Value threadId = getThreadId(rewriter, loc);
+  auto mod = op->getParentOfType<ModuleOp>();
+  unsigned iWarpSize = triton::gpu::TritonGPUDialect::getThreadsPerWarp(mod);
+  Value warpSize = i32_val(iWarpSize);
+  Value warpId = udiv(threadId, warpSize);
+  Value laneId = urem(threadId, warpSize);
+
   auto [laneIdAxis, warpIdAxis, flatIdParallel] =
-      getDelinearizedIds(rewriter, helper);
+      getDelinearizedIds(rewriter, helper, laneId, warpId);
```

Comment:
Do you plan to upstream this part?
They could be interested in elimination of hardcoded warp sizes.

Reply:
For sure.

Reply:
- https://github.com/openai/triton/pull/2272